src/PowerShellEditorServices.Engine/Services/PowerShellContext/Utilities/CommandHelpers.cs

```diff
@@ -0,0 +1,111 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using Microsoft.PowerShell.EditorServices.Utility;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using System.Threading.Tasks;
+
+namespace Microsoft.PowerShell.EditorServices
+{
+    /// <summary>
+    /// Provides utility methods for working with PowerShell commands.
+    /// </summary>
+    public static class CommandHelpers
+    {
+        private static readonly HashSet<string> NounBlackList =
```

Comment:
I suspect this isn't threadsafe and should be a `ConcurrentDictionary<string, bool>` or something, where the value is never used

Reply:
We should also change `blacklist` to `exclusionlist`

Reply:
done